Bug/3840 --forbid-only doesn't recognize `it.only` when `before` crashes

rpgeeganage opened the pull request:
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions.

### Description of the Change

`--forbid-only` doesn't recognize `it.only` when `before` crashes.

### Alternate Designs

<!-- Explain what other alternates were considered and why the proposed version was selected -->

### Why should this be in core?
This is an attempt to solve the above-mentioned bug.
<!-- Explain why this functionality should be in mocha as opposed to its own package -->

### Benefits
This is an attempt to solve the above-mentioned bug.

<!-- What benefits will be realized by the code change? -->

### Possible Drawbacks
None
<!-- What are the possible side-effects or negative impacts of the code change? -->

### Applicable issues
Fixes #3840 

* Is this a breaking change (major release)? **No**
* Is it an enhancement (minor release)? **No**
* Is it a bug fix, or does it not impact production code (patch release)? **Yes**

juergba commented:
@rpgeeganage thanks for this PR. 

Some thoughts:
- You are supposing that `.only` has precedence over `skip()`. Have you found any evidence for this in the docu or code base? BTW I don't know the answer.
- Your solution works for `describe.skip()`and `before(... this.skip())`, but fails for `beforeEach(... this.skip())`
- please rename your test and put it in the directory "test\integration\fixtures\options\forbid-only". It's difficult not to loose overview in a sea of tests named with regression numbers.

rpgeeganage commented:
@boneskull , @juergba ,
sorry for the late reply. I may have screwed up something. I'll look into it.

rpgeeganage commented:
@boneskull , @juergba 
I have opened a new PR (https://github.com/mochajs/mocha/pull/3948). 
Please check. I have moved the tests and changed the fix to work with `before` and `beforeEach`.
Sorry for unable to open this PR.
